src/main/scala/sbtghactions/JavaSpec.scala

```diff
@@ -18,15 +18,58 @@ package sbtghactions
 
 final case class JavaSpec(dist: JavaSpec.Distribution, version: String) {
   def render: String = dist match {
-    case JavaSpec.Distribution.GraalVM(gversion) => s"graal_$gversion@$version"
+    case JavaSpec.Distribution.GraalVM(Graalvm.Version(gversion)) =>
+      s"graal_$gversion@$version"
+    case JavaSpec.Distribution.GraalVM(Graalvm.Distribution(distribution)) =>
+      s"graal_$distribution@$version"
     case dist => s"${dist.rendering}@$version"
   }
 }
 
+/**
+ * @see https://github.com/graalvm/setup-graalvm#migrating-from-graalvm-223-or-earlier-to-the-new-graalvm-for-jdk-17-and-later
+ */
+sealed trait Graalvm {
+  private[sbtghactions] def compile: String
```

Comment:
This is private because we don't want to expose this method to sbt build's in case we need to change it for later.